File: game-rules/src/Rules/GameRulesWorkflowInterface.php
<?php

declare(strict_types=1);

namespace Temporal\SnakeGame\Rules;

use Temporal\Workflow\QueryMethod;
use Temporal\Workflow\WorkflowInterface;
use Temporal\Workflow\WorkflowMethod;
use Temporal\Workflow\SignalMethod;

#[WorkflowInterface]
interface GameRulesWorkflowInterface
{
    #[WorkflowMethod(name: "BoardRules.new")]
    public function exec(
        int $allDots, int $dotSize
    );

    #[QueryMethod]
    public function getX(): array;

    #[QueryMethod]
    public function getY(): array;

    #[QueryMethod]
    public function getDots(): int;

    #[SignalMethod]
    public function init(): void;

    #[SignalMethod]
    public function move(): void;

    #[SignalMethod]
    public function moveLeft(): void;

    #[SignalMethod]
    public function moveRight(): void;

    #[SignalMethod]
    public function moveUp(): void;

    #[SignalMethod]
    public function moveDown(): void;

    #[SignalMethod]
    public function addDot(): void;

    #[SignalMethod]
    public function exitGame(): void;

}

File: game/src/main/java/io/temporal/snakegame/GameRulesWorkflowInterface.java
package io.temporal.snakegame;

import io.temporal.workflow.QueryMethod;
import io.temporal.workflow.SignalMethod;
import io.temporal.workflow.WorkflowInterface;
import io.temporal.workflow.WorkflowMethod;

@WorkflowInterface
public interface GameRulesWorkflowInterface {
    // FIXME: Node SDK uses workflow filename as name
    @WorkflowMethod(name = "GameRulesWorkflow")
    // FIXME: Node SDK requires main function
    void main(int allDots, int dotSize);

    @QueryMethod
    int[] getX();

    @QueryMethod
    int[] getY();

    @QueryMethod
    int getDots();

    @SignalMethod
    void init();

    @SignalMethod
    void move();

    @SignalMethod
    void moveLeft();

    @SignalMethod
    void moveRight();

    @SignalMethod
    void moveUp();

    @SignalMethod
    void moveDown();

    @SignalMethod
    void addDot();

    @SignalMethod
    void exitGame();
}
